### PR TITLE
Add support for more platforms in OSGi testing

### LINKED ISSUE #347: JavaCPP fails tests on ARM64
Trying to build on ARM64

Note the x86_64 in the error log ...
Native library not found in JAR: org/bytedeco/javacpp/test/osgi/linux-x86_64/libjniCalc.so
Below

Note (2) a quick look around reveals libjniCalc.so on...
./target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/linux-arm64/libjniCalc.so

Edited src/it/osgi/pom.xml and changed the path - build worked (hackety, hack, hack(

===== Log =====

[INFO] [INFO] ------------------< org.bytedeco:javacpp-tests-osgi >-------------------
[INFO] [INFO] Building JavaCPP OSGi Tests 1.5.2-SNAPSHOT
[INFO] [INFO] --------------------------------[ jar ]---------------------------------
[INFO] [INFO]
[INFO] [INFO] --- javacpp:1.5.2-SNAPSHOT:build (javacpp.parser) @ javacpp-tests-osgi ---
[INFO] [WARNING] No classes found in package org.bytedeco.javacpp.test.osgi
[INFO] [INFO] Detected platform "linux-arm64"
[INFO] [INFO] Building for platform "linux-arm64"
[INFO] [INFO]
[INFO] [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ javacpp-tests-osgi ---
[INFO] [WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] [INFO] skip non existing resourceDirectory /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/src/main/resources
[INFO] [INFO]
[INFO] [INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ javacpp-tests-osgi ---
[INFO] [INFO] Changes detected - recompiling the module!
[INFO] [WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] [INFO] Compiling 2 source files to /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes
[INFO] [INFO]
[INFO] [INFO] --- javacpp:1.5.2-SNAPSHOT:build (javacpp.compiler) @ javacpp-tests-osgi ---
[INFO] [INFO] Detected platform "linux-arm64"
[INFO] [INFO] Building for platform "linux-arm64"
[INFO] [WARNING] Could not load platform properties for class org.bytedeco.javacpp.test.osgi.JavaCPPOsgiTest
[INFO] [INFO] Generating /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/jnijavacpp.cpp
[INFO] [INFO] Generating /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/jniCalc.cpp
[INFO] [INFO] Compiling /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/linux-arm64/libjniCalc.so
[INFO] [INFO] g++ -I/home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/src/main/c -I/usr/lib/jvm/java-8-openjdk-arm64/include -I/usr/lib/jvm/java-8-openjdk-arm64/include/linux /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/jniCalc.cpp /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/jnijavacpp.cpp -O3 -s -Wl,-rpath,$ORIGIN/ -Wl,-z,noexecstack -Wl,-Bsymbolic -Wall -fPIC -pthread -shared -o libjniCalc.so
[INFO] [INFO] Deleting /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/org/bytedeco/javacpp/test/osgi/jniCalc.cpp
[INFO] [INFO] Deleting /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/target/classes/jnijavacpp.cpp
[INFO] [INFO]
[INFO] [INFO] --- bnd-maven-plugin:4.2.0:bnd-process (default) @ javacpp-tests-osgi ---
[INFO] [ERROR] /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/pom.xml [0:0]: Native library not found in JAR: org/bytedeco/javacpp/test/osgi/linux-x86_64/libjniCalc.so
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] BUILD FAILURE
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] Total time:  11.117 s
[INFO] [INFO] Finished at: 2019-10-03T16:19:18+01:00
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [ERROR] Failed to execute goal biz.aQute.bnd:bnd-maven-plugin:4.2.0:bnd-process (default) on project javacpp-tests-osgi: Native library not found in JAR: org/bytedeco/javacpp/test/osgi/linux-x86_64/libjniCalc.so -> [Help 1]
[INFO] [ERROR]
[INFO] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[INFO] [ERROR] Re-run Maven using the -X switch to enable full debug logging.
[INFO] [ERROR]
[INFO] [ERROR] For more information about the errors and possible solutions, please read the following articles:
[INFO] [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[INFO]   The build exited with code 1. See /home/simon/src/jcpp/master/javacpp/target/integration-test/projects/osgi/build.log for details.
[INFO]           osgi/pom.xml ..................................... FAILED (15.2 s)
[INFO] -------------------------------------------------
[INFO] Build Summary:
[INFO]   Passed: 0, Failed: 1, Errors: 0, Skipped: 0
[INFO] -------------------------------------------------
[ERROR] The following builds failed:
[ERROR] *  osgi/pom.xml
[INFO] -------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:07 min
[INFO] Finished at: 2019-10-03T16:19:18+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-invoker-plugin:3.2.0:run (integration-test) on project javacpp: 1 build failed. See console output above for details. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

### PR DESCRIPTION
**Pull request by timothyjward**
The list of supported platforms for OSGi testing isn't intended to be exhaustive, as at some point we stop testing JavaCPP and start testing Maven/OSGi, however it does need to reflect the platforms that JavaCPP is built on. The list has therefore been extended to cover windows, and a wider variety of CPU archetectures on Linux.

Android and iOS have been ignored deliberately as the Maven build tool doesn't natively support them. They could be added in future if this requirement changes

Signed-off-by: Tim Ward <timothyjward@apache.org>

This PR should fix #347

### CODE REVIEW

**Comment by saudet** [comment]
Should do for now, but at some point it'll probably not be worth using immature features from OSGi.

File: src/it/osgi/pom.xml [hunk 1]
```diff
@@ -13,6 +13,18 @@
   </properties>
   
   <profiles>
```
Comment (saudet) [inline src/it/osgi/pom.xml]: There's a few missing or incorrect ones, use these ones for consistency at least:
https://github.com/bytedeco/javacpp-presets/blob/master/pom.xml#L1914
The name of the DLL on Windows is also incorrect.

**Comment by timothyjward** [comment]
I'm hoping that the latest commit is what you had in mind, it's certainly more elegant than what was there before. I decided not to try to make the preset pom the parent of the test pom as it would have introduced a horrible dependency cycle into JavaCPP, as well as bringing a lot of other preset profiles into the test.

**Comment by saudet** [comment]
Yes, that works, that's what I had in mind. But this does make me wonder what we should be doing for the `jnijavacpp` library I started talking about in pull #344. Do we want to move that parent `pom.xml` file to this repository and make the presets depend on it instead? It would help anyone wanting to do something like the presets without a weird dependency on the presets themselves.

**Comment by saudet** [comment]
This still doesn't fix the test on Windows though. OSGi fails with the following error, whatever that means:
```
[INFO] [INFO] --- bnd-testing-maven-plugin:4.2.0:testing (default) @ javacpp-tests-osgi ---
[INFO] [INFO] Matching glob *
[INFO] ! could not resolve the bundles: [javacpp-tests-osgi-1.5.2.201910050921 org.osgi.framework.BundleException: Unable to resolve javacpp-tests-osgi [2](R 2.0): missing requirement [javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows)) Unresolved requirements: [[javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows))]
[INFO] ]
[INFO] ! Failed to start bundle javacpp-tests-osgi-1.5.2.201910050921, exception Unable to resolve javacpp-tests-osgi [2](R 2.0): missing requirement [javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows)) Unresolved requirements: [[javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows))]
[INFO] org.osgi.framework.BundleException: Unable to resolve javacpp-tests-osgi [2](R 2.0): missing requirement [javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows)) Unresolved requirements: [[javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows))]
[INFO]  at org.apache.felix.framework.Felix.resolveBundleRevision(Felix.java:4368)
[INFO]  at org.apache.felix.framework.Felix.startBundle(Felix.java:2281)
[INFO]  at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:998)
[INFO]  at aQute.launcher.Launcher.startBundles(Launcher.java:528)
[INFO]  at aQute.launcher.Launcher.activate(Launcher.java:427)
[INFO]  at aQute.launcher.Launcher.run(Launcher.java:306)
[INFO]  at aQute.launcher.Launcher.main(Launcher.java:152)
[INFO] Unable to resolve javacpp-tests-osgi [2](R 2.0): missing requirement [javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows)) Unresolved requirements: [[javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows))]
[INFO] TEST testAllResolved(aQute.junit.UnresolvedTester) <<< FAILURE: Unresolved bundles
[INFO] javacpp-tests-osgi [2];1.5.2.201910050921
[INFO]     Unable to resolve javacpp-tests-osgi [2](R 2.0): missing requirement [javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows)) Unresolved requirements: [[javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows))]
[INFO]
[INFO]
[INFO] junit.framework.AssertionFailedError: Unresolved bundles
[INFO] javacpp-tests-osgi [2];1.5.2.201910050921
[INFO]     Unable to resolve javacpp-tests-osgi [2](R 2.0): missing requirement [javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows)) Unresolved requirements: [[javacpp-tests-osgi [2](R 2.0)] osgi.native; (&(osgi.native.osname~=windows))]
[INFO]
[INFO]
[INFO]  at junit.framework.Assert.fail(Assert.java:57)
[INFO]  at junit.framework.Assert.assertTrue(Assert.java:22)
[INFO]  at junit.framework.TestCase.assertTrue(TestCase.java:192)
[INFO]  at aQute.junit.UnresolvedTester.testAllResolved(UnresolvedTester.java:51)
[INFO]  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[INFO]  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[INFO]  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[INFO]  at java.lang.reflect.Method.invoke(Method.java:498)
[INFO]  at junit.framework.TestCase.runTest(TestCase.java:176)
[INFO]  at junit.framework.TestCase.runBare(TestCase.java:141)
[INFO]  at junit.framework.TestResult$1.protect(TestResult.java:122)
[INFO]  at junit.framework.TestResult.runProtected(TestResult.java:142)
[INFO]  at junit.framework.TestResult.run(TestResult.java:125)
[INFO]  at junit.framework.TestCase.run(TestCase.java:129)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:252)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:247)
[INFO]  at junit.framework.TestSuite.runTest(TestSuite.java:252)
[INFO]  at junit.framework.TestSuite.run(TestSuite.java:247)
[INFO]  at aQute.junit.Activator.test(Activator.java:340)
[INFO]  at aQute.junit.Activator.run(Activator.java:158)
[INFO]  at aQute.launcher.Launcher.lambda$serviceChanged$0(Launcher.java:1385)
[INFO]  at aQute.launcher.Launcher.run(Launcher.java:352)
[INFO]  at aQute.launcher.Launcher.main(Launcher.java:152)
```

**Comment by timothyjward** [comment]
The issue here is that for some reason the requirement for "windows" in the tester bundle isn't being matched (which is what `osgi.native; (&(osgi.native.osname~=windows))` means).

The ~= is case insensitive, but I apparently I misremembered `windows` as the alias for all of windows. I should have used `Win32`, serves me right for not checking the reference page! https://www.osgi.org/developer/specifications/reference/#ReferenceOS

**Pull request merged**